File: Checkmate Game C++/LocalPlayer.cpp
/*
 * LocalPlayer.cpp
 *
 */

#include "LocalPlayer.h"
using namespace std;

LocalPlayer::LocalPlayer(const int &symbol) :
		Player(symbol), player_symbol_(empty), opponent_symbol_(empty), player_turn_(
				false)
{
	setPlayerSymbol(symbol);
	if (symbol == player_1)
	{
		setOpponentSymbol(player_2);
	}
	else if (symbol == player_2)
	{
		setOpponentSymbol(player_1);
	}
}

const int LocalPlayer::getPlayerSymbol() const
{
	return (player_symbol_);
}

void LocalPlayer::setPlayerSymbol(const int &player_symbol)
{
	player_symbol_ = player_symbol;
}

bool LocalPlayer::isPlayerTurn() const
{
	return (player_turn_);
}

void LocalPlayer::setPlayerTurn(const bool &player_turn)
{
	player_turn_ = player_turn;
}

const int LocalPlayer::getOpponentSymbol() const
{
	return (opponent_symbol_);
}

void LocalPlayer::setOpponentSymbol(const int &opponent_symbol)
{
	opponent_symbol_ = opponent_symbol;
}

LocalPlayer::~LocalPlayer()
{
}

File: Reversi Game Java/src/LocalPlayer.java
/**
 * LocalPlayer class - the game local player.
 *
 */
public class LocalPlayer implements Player {

	private int player_symbol_ = GamePieceSymbol.EMPTY.getSymbolValue(); // Default value
	private int opponent_symbol_ = GamePieceSymbol.EMPTY.getSymbolValue(); // Default value
	private Boolean player_turn_ = false; // Default value

	/**
	 * Function Name: LocalPlayer
	 * Description: construct player with given symbol.
	 * 
	 * @param symbol- the symbol to set as player symbol.
	 * @return: void
	 */
	public LocalPlayer(int symbol)

	{
		setPlayerSymbol(symbol);
		if (symbol == GamePieceSymbol.PLAYER_1.getSymbolValue()) {
			setOpponentSymbol(GamePieceSymbol.PLAYER_2.getSymbolValue());
		} else if (symbol == GamePieceSymbol.PLAYER_2.getSymbolValue()) {
			setOpponentSymbol(GamePieceSymbol.PLAYER_1.getSymbolValue());
		}
	}

	public int getPlayerSymbol() {
		return (this.player_symbol_);
	}

	public void setPlayerSymbol(int player_symbol) {
		this.player_symbol_ = player_symbol;
	}

	public boolean isPlayerTurn() {
		return (this.player_turn_);
	}

	public void setPlayerTurn(boolean player_turn) {
		this.player_turn_ = player_turn;
	}

	public int getOpponentSymbol() {
		return (this.opponent_symbol_);
	}

	public void setOpponentSymbol(int opponent_symbol) {
		this.opponent_symbol_ = opponent_symbol;
	}
}
